clang/test/SemaSYCL/stall_enable.cpp

```diff
@@ -0,0 +1,48 @@
+// RUN: %clang_cc1 %s -fsyntax-only -fsycl -internal-isystem %S/Inputs -fsycl-is-device -Wno-sycl-2017-compat -DTRIGGER_ERROR -verify
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -internal-isystem %S/Inputs -fsyntax-only -ast-dump -Wno-sycl-2017-compat %s | FileCheck %s
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+queue q;
+
+#ifdef TRIGGER_ERROR
+[[intel::stall_enable(1)]] void bar1() {} // expected-error{{'stall_enable' attribute takes no arguments}}
+[[intel::stall_enable]] int N;            // expected-error{{'stall_enable' attribute only applies to functions}}
+#endif
+
+struct FuncObj {
+  [[intel::stall_enable]] void operator()() const {}
+};
+
+[[intel::stall_enable]] void func_do_not_ignore() {}
```

Comment:
Likewise, this one too.  Is the attribute valid only on lambdas and function objects passed to the kernel or even on functions called from the kernel?

Reply:
The attribute is valid only on lambdas and function objects passed to the kernel not on function called from kernel.

Reply:
If the attribute is applied to a function that the kernel calls, it should produce a syntax error.